### PR TITLE
document rhcs scripts

### PR DESCRIPTION
**Pull request by ktdreyer**
Document how to build the RH Ceph Storage container image locally.

### CODE REVIEW

File: contrib/rhcs.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+**Note: this uses an internal Pulp server, so your computer must be on Red
+Hat's network to access these repositories.**
+
+## Running a build
+
+```
+cd staging/pacific-ubi8-latest-x86_64/composed
+make build DAEMON_BASE_IMAGE=test
```
Comment (tshacked) [inline contrib/rhcs.md:82]: @ktdreyer, thanks for the detailed instructions.
I assume that the created image can be used to run any ceph daemon.
- Is there a guide on how to run each daemon? this [wiki](https://github.com/ceph/ceph-container/wiki) is not full and maybe outdated.
- Or maybe I should use [`cephadm`](https://docs.ceph.com/en/quincy/cephadm/index.html) - is there a way to use it with local image?

**Review by tshacked — APPROVED** [review]

File: contrib/rhcs.md (hunk 1 above)
Comment (ktdreyer) [reply contrib/rhcs.md:82]: @guits would you please assist @tshacked with the proper way to run an image she's built locally?

Comment (guits) [reply contrib/rhcs.md:82]: sure

File: contrib/rhcs.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+You will then be able to run `podman pull registry.redhat.io/ubi8/ubi:latest`
+to fetch the base image.
+
+## Composing the Dockerfile
+
+The `ceph-container` project uses a series of template files to create the
+final `Dockerfile` that developers can build or commit to dist-git. This
+command generates that downstream Red Hat UBI-based `Dockerfile`:
```
Comment (tshacked) [inline contrib/rhcs.md:21]: nit: mention that it should be done on a stable-x.0 branch

File: contrib/rhcs.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+We use [OSBS](https://osbs.readthedocs.io/en/osbs_ocp3/) to build the images for RH storage products. OSBS dynamically injects Yum repository definition steps into our `Dockerfile` during each build. This gives us far more flexibility (for hotfixes, etc) than storing all repo files in Git.
+
+To test a local build outside OSBS, you must mimic what OSBS does and insert some Yum repo definitions.
+
+After the `RUN rm -f /etc/yum.repos.d/ubi.repo` step, add the following by-hand steps to your `staging/pacific-ubi8-latest-x86_64/composed/Dockerfile`:
+
+```Dockerfile
+RUN printf '[rhel-8-baseos]\n\
+name = Red Hat Enterprise Linux 8 BaseOS\n\
+baseurl = http://rhsm-pulp.corp.redhat.com/content/dist/rhel8/8/$basearch/baseos/os/\n\
```
[the remark this answers is not in the document]
Comment (ktdreyer) [reply contrib/rhcs.md:40]: Exactly, see the instructions about the VPN.

[the remark this answers is not in the document]
Comment (ktdreyer) [reply contrib/rhcs.md:40]: "**Note: this uses an internal Pulp server, so your computer must be on Red Hat's network to access these repositories.**"

File: contrib/rhcs.md (hunk 2 above)
Comment (ktdreyer) [reply contrib/rhcs.md:23]: Thanks for the suggestion. I've updated it to recommend `stable-6.0` since that corresponds to RH Ceph Storage 5.

**Comment by guits** [comment]
jenkins test ceph_ansible-collocation

**Pull request merged**